gcsfs/core.py

```diff
@@ -124,24 +124,30 @@ class GCSFileSystem(object):
     """
     Connect to Google Cloud Storage.
 
-    Two modes of authentication are supported:
-    - if ``token=None``, GCSFS will attempt to use your default google
-      credentials; if this is not found, you will be given a "device code",
-      which you must enter into a browser where you are logged in with your
-      Google identity.
+    The following modes of authentication are supported:
+    - ``token=None``, GCSFS will attempt to guess your credentials in the
+      following order: gcloud CLI default, gcsfs cached token, google compute
+      metadata service, anonymous.
+    - ``token='gtoken'``, your default gcloud credentials will be used, which
+      are typically established by doing ``gcloud login`` in a terminal.
+    - ``token=='cache'``, credentials from previoudly successful gcsfs
+      authentication will be used (use this after "browser" auth succeeded)
+    - ``token='anon'``, no authentication is preformed, and you can only
+      access data which is accessible to allUsers (in this case, the project and
+      access level parameters are meaningless)
+    - ``token='browser'``, you get an access code with which you can
+      authenticate via a specially provided URL
     - if ``token='cloud'``, we assume we are running within google compute
       or google container engine, and query the internal metadata directly for
       a token.
     - you may supply a token generated by the
       [gcloud](https://cloud.google.com/sdk/docs/)
       utility; this is either a python dictionary, or the name of a file
-      containing the JSON returned by logging in with the gcloud CLI tool. On
-      a posix system this may be at
-      ``c``
-
-    We maintain a cache of refresh tokens in the file ~/.gcs_tokens, so for any
-    pair of (project, access), you will not need to log in once your credentials
-    are verified.
+      containing the JSON returned by logging in with the gcloud CLI tool.
+      This may be at
+      ``~/.config/gcloud/application_default_credentials.json``,
+      `` ~/.config/gcloud/credentials``, or
+      ``~\AppData\Roaming\gcloud\credentials``, etc.
```

Comment:
Might we also provide the token explicitly as a dictionary?

Reply:
Yes, you can provide a dictionary - it does say that, but could provide an example.